src/Microsoft.ML.Core/Environment/HostEnvironmentBase.cs

```diff
@@ -460,98 +452,6 @@ protected virtual IProgressChannel StartProgressChannelCore(HostBase host, strin
-            return new SimpleFileHandle(env, path, needsWrite: false, autoDelete: false);
-        }
-
-        public IFileHandle CreateOutputFile(string path)
-        {
-            if (Master != null)
-                return Master.CreateOutputFileCore(this, path);
-            return CreateOutputFileCore(this, path);
-        }
-
-        /// <summary>
-        /// This "creates" the output file handle. The default implementation assumes the file resides in the
-        /// file system and it does not create the physical file until a write stream is opened on the returned
-        /// file handle.
-        /// </summary>
-        protected virtual IFileHandle CreateOutputFileCore(IHostEnvironment env, string path)
-        {
-            this.AssertValue(env);
-            this.CheckNonWhiteSpace(path, nameof(path));
-            if (Master != null)
-                return Master.CreateOutputFileCore(env, path);
-            return new SimpleFileHandle(env, path, needsWrite: true, autoDelete: false);
-        }
-
-        public IFileHandle CreateTempFile(string suffix = null, string prefix = null)
-        {
-            if (Master != null)
-                return Master.CreateAndRegisterTempFile(this);
-            return CreateAndRegisterTempFile(this);
-        }
-
-        /// <summary>
-        /// This calls CreateTempFileCore and handles registering the temp file for cleanup when the environment is disposed.
-        /// </summary>
-        protected IFileHandle CreateAndRegisterTempFile(IHostEnvironment env, string suffix = null, string prefix = null)
-        {
-            this.AssertValue(env);
-
-            if (Master != null)
-                return Master.CreateAndRegisterTempFile(env, suffix, prefix);
-
-            var file = CreateTempFileCore(env, suffix, prefix);
-
-            lock (_tempLock)
-            {
-                if (_tempFiles == null)
-                {
-                    file.Dispose();
-                    throw env.Except("This environment has been disposed, so can't allocate new temp files");
-                }
-                _tempFiles.Add(file);
-            }
-
-            return file;
-        }
-
-        protected virtual IFileHandle CreateTempFileCore(IHostEnvironment env, string suffix = null, string prefix = null)
-        {
-            this.CheckParam(!HasBadFileCharacters(suffix), nameof(suffix));
-            this.CheckParam(!HasBadFileCharacters(prefix), nameof(prefix));
-
-            Guid guid = Guid.NewGuid();
-            string path = Path.GetFullPath(Path.Combine(Path.GetTempPath(), prefix + guid.ToString() + suffix));
-            return new SimpleFileHandle(env, path, needsWrite: true, autoDelete: true);
-        }
-
-        /// <summary>
-        /// Returns true if the given string is non-null and contains invalid file name characters.
-        /// </summary>
-        protected virtual bool HasBadFileCharacters(string str = null)
```

Comment:
Are we still doing these checks on the new Util method?

Reply:
comment for next PR